storage/innobase/include/log0log.h

```diff
@@ -33,6 +33,17 @@ Created 12/9/1995 Heikki Tuuri
 #include "srw_lock.h"
 #include <string>
 
+/* According to Linux "man 2 read" and "man 2 write" this is the
+maximum size of a read or write request, both on 32-bit and 64-bit
+systems.
+
+On FreeBSD, the limit is close to the Linux one, INT_MAX.
+
+On Microsoft Windows, the limit is 4 GiB - 1.
+
+On other systems, the limit typically is up to SSIZE_T_MAX. */
+static constexpr unsigned innodb_log_buffer_size_max= 0x7ffff000;
```

Comment:
maybe name it single_io_max_size or something that better fits the comment, and put it into os0file.h, where SyncFileIO is defined? The fact. that log buffer is used for the IO, this can perhaps change in the future, however Linux limitation is  going to remain forever.

Reply:
0892e6d028f7b08da938defd9443ecbb45cda612 defines `constexpr unsigned os_file_request_size_max` and aliases it as `log_sys.buf_size_max`.